### PR TITLE
No dom query in grid element resize strategy

### PR DESCRIPTION
**Pull request by gbalint**
**Problem:**
Another step to remove dom api dependency from grid strategies: the grid-resize-element strategy uses the new `getGlobalFramesOfGridCells` to detect the grid cell under the mouse.

**Commit Details:** 

- Fixed `getGlobalFramesOfGridCells` problem: it calculated the local frame not the global frame....
- Fixed and improved `getGlobalFramesOfGridCells` test to include differend padding/row-padding/column-padding values
- Memoized `getGlobalFramesOfGridCells`: this is a quite fast function so it wouldn't be important to memoize it, but in most cases we run it for the same grid cell (because we are doing an interaction in it), so it still makes sense to memoize. I just uses `memoize` with the default config (which allows memoization of 5 values)
- Created a new `getGridCellUnderMouse` version called `getGridCellUnderMouseFromMetadata`, and call that from the strategy
- I modified the strategy to use the canvas and not the window coordinate system. Unfortunately, `GridCustomStrategyState.targetCellData` uses the window coordinate system, and that is used in multiple strategies, so I postponed to change that too. I introduced a new type `TargetGridCellDataCanvas` (similar to `TargetGridCellData`), I plan to migrate to this new type everywhere.

**TODO:**

The strategy is still not the best. A downside of querying the grid cells under the mouse is that the strategy is broken when the mouse is not over the grid (even though the resize rectangle follows the mouse), see https://screenshot.click/16-30-s1avn-hgu83.mp4

Using the metadata it is possible to implement a much better strategy which can calculate where to resize to even when the mouse is not over the grid. That can come in a subsequent PR.

**Manual Tests:**
I hereby swear that:

- [x] I opened a hydrogen project and it loaded
- [x] I could navigate to various routes in Preview mode

### CODE REVIEW

File: editor/src/components/canvas/canvas-strategies/strategies/grid-cell-bounds.ts [hunk 1]
```diff
@@ -27,6 +33,38 @@ export function gridCellTargetId(
...
+  const gridCellGlobalFrames = getGlobalFramesOfGridCells(grid)
+
+  if (gridCellGlobalFrames == null) {
+    return null
+  }
+
+  return getGridCellUnderPoint(gridCellGlobalFrames, canvasPoint)
+}
+
+// TODO: optimization: binary search
```
Comment (Rheeseyb) [inline editor/src/components/canvas/canvas-strategies/strategies/grid-cell-bounds.ts:49]: Unless we have a grid with a ton of rows / columns this really won't matter

File: editor/src/components/canvas/dom-lookup.ts [hunk 2]
```diff
@@ -636,3 +639,18 @@ export function canvasPointToWindowPoint(
     throw new Error('calling screenToElementCoordinates() before being mounted')
   }
 }
+
+export function canvasRectangleToWindowRectangle(
```
Comment (Rheeseyb) [inline editor/src/components/canvas/dom-lookup.ts:643]: Huh, I thought we already had this function

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/canvas/canvas-strategies/strategies/grid-cell-bounds.ts [hunk 3]
```diff
@@ -27,6 +33,38 @@ export function gridCellTargetId(
...
+          gridCellCoordinates: gridCellCoordinates(i + 1, j + 1),
+          cellCanvasRectangle: gridCellGlobalFrames[i][j],
+        }
+      }
+    }
+  }
+  return null
+}
+
+// TODO: should be superseded by getGridCellUnderMouseNoDomLookup
```
Comment (balazsbajorics) [inline editor/src/components/canvas/canvas-strategies/strategies/grid-cell-bounds.ts:67]: do you mean

```suggestion
// TODO: should be superseded by getGridCellUnderMouseFromMetadata
```


?

**Review by liady — APPROVED** [review]

**Review by balazsbajorics — APPROVED** [review]

File: editor/src/components/canvas/dom-lookup.ts (hunk 2 above)
Comment (gbalint) [reply editor/src/components/canvas/dom-lookup.ts:643]: me too :)

File: editor/src/components/canvas/canvas-strategies/strategies/grid-cell-bounds.ts (hunk 3 above)
Comment (gbalint) [reply editor/src/components/canvas/canvas-strategies/strategies/grid-cell-bounds.ts:67]: yes, meanwhile I renamed the function :)

**Pull request merged**